module/motion/OldWalkEngine/src/FootFallPlanner.cpp

```diff
@@ -70,46 +80,53 @@ namespace motion {
 
         uTorsoDestination = stepTorso(uLeftFootDestination, uRightFootDestination, 0.5);
 
+        Eigen::Affine2d supportModTranslate = Eigen::Affine2d::Identity();
+        supportModTranslate.translation()   = supportMod;
+
+        Eigen::Affine2d footOffsetNeg = Eigen::Affine2d::Identity();
+        footOffsetNeg.translation()   = -footOffset;
+
         // apply velocity-based support point modulation for uSupport
         if (swingLeg == LimbID::RIGHT_LEG) {
-            Transform2D uLeftFootTorso  = uTorsoSource.worldToLocal(uLeftFootSource);
-            Transform2D uTorsoModded    = uTorso.localToWorld({supportMod[0], supportMod[1], 0});
-            Transform2D uLeftFootModded = uTorsoModded.localToWorld(uLeftFootTorso);
-            uSupport                    = uLeftFootModded.localToWorld({-footOffset[0], -footOffset[1], 0});
+            Eigen::Affine2d uLeftFootTorso  = worldToLocal(uTorsoSource, uLeftFootSource);
+            Eigen::Affine2d uTorsoModded    = localToWorld(uTorso, supportModTranslate);
+            Eigen::Affine2d uLeftFootModded = localToWorld(uTorsoModded, uLeftFootTorso);
+            uSupport                        = localToWorld(uLeftFootModded, footOffsetNeg);
         }
         else {
-            Transform2D uRightFootTorso  = uTorsoSource.worldToLocal(uRightFootSource);
-            Transform2D uTorsoModded     = uTorso.localToWorld({supportMod[0], supportMod[1], 0});
-            Transform2D uRightFootModded = uTorsoModded.localToWorld(uRightFootTorso);
-            uSupport                     = uRightFootModded.localToWorld({-footOffset[0], footOffset[1], 0});
+            Eigen::Affine2d uRightFootTorso = worldToLocal(uTorsoSource, uRightFootSource);
+
```

Comment:
```suggestion
```

Reply:
As in, eigenify the commented out stuff? could you please clarify what you mean?

Reply:
Just delete the blank line :P